axiom/nr_php_packages.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef nr_php_packages_HDR
+#define nr_php_packages_HDR
+
+#include "nr_log_event.h"
+#include "util_random.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+
+typedef struct _nr_php_package_t {
+  char* package_name;
+  char* package_version;
+} nr_php_package_t;
+
+typedef nr_hashmap_t nr_php_packages_t;
+
+/*
+ * Purpose : Create a new php package
+ *
+ * Params  : 1. Package name
+ *           2. Package version
+ * 
+ * Returns : A php package that has a name and version. If nr_php_packages_add_package()
+ *           is not called, then it must be freed by nr_php_package_destroy()
+ */
+extern nr_php_package_t* nr_php_package_create(char* name, char* version);
+
+/*
+ * Purpose : Destroy/free php package
+ *
+ * Params  : The php package to free
+ * 
+ * Returns : Nothing, it is void
+ */
+extern void nr_php_package_destroy(nr_php_package_t* p);
```

Comment:
I suggest `nr_php_package_destroy` to set `p` to `NULL`:
```suggestion
extern void nr_php_package_destroy(nr_php_package_t** p);
```

Reply:
After discussion, this will stay the same. Explanation here: https://github.com/newrelic/newrelic-php-agent/pull/754#discussion_r1376736594